### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by sfdez0**
The pull request includes both task 6 and 7.

Old pull requests:
task 6 - #49 
task 7 - #50

### CODE REVIEW

File: Assignment4/sfdez0-a180439/task07.py
Comment (anaigmo) [inline Assignment4/sfdez0-a180439/task07.py:1]: RDFlib alternatives for tasks 7.1-3 are missing

File: Assignment4/sfdez0-a180439/task07.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+from rdflib import FOAF
+
+q4 = prepareQuery('''
+  SELECT ?Person WHERE {
+    ?Person foaf:knows ns:RockySmith
+  }
+  ''',
+  initNs = { "ns": Namespace("http://somewhere#"), "foaf": FOAF }
+)
```
Comment (anaigmo) [inline Assignment4/sfdez0-a180439/task07.py]: list the name, not the iri!

File: Assignment4/sfdez0-a180439/task07.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+from rdflib import FOAF
+
+q5 = prepareQuery('''
+  SELECT DISTINCT ?Entity WHERE {
+    ?Entity foaf:knows ?Entity2;
+            foaf:knows ?Entity3
+  }
+  ''',
+  initNs = { "foaf": FOAF }
```
Comment (anaigmo) [inline Assignment4/sfdez0-a180439/task07.py]: There is. nothing avoiding ?Entity2 and ?Entity3 from being the same, this query would list all triples that use foaf:knows, but results with one will appear too

File: Assignment4/sfdez0-a180439/task07.py
Comment (sfdez0) [reply Assignment4/sfdez0-a180439/task07.py:1]: Fixed with commit adc5377

File: Assignment4/sfdez0-a180439/task07.py (hunk 1 above)
Comment (sfdez0) [reply Assignment4/sfdez0-a180439/task07.py]: Fixed with commit fe3a863

File: Assignment4/sfdez0-a180439/task07.py (hunk 2 above)
Comment (sfdez0) [reply Assignment4/sfdez0-a180439/task07.py]: Fixed with commit 936b664

**Pull request merged**